lib/src/flutter_secure_file_storage.dart

```diff
@@ -130,11 +143,25 @@ class FlutterSecureFileStorage {
 
   Future<void> _updateKeys() async {
     final encodedData = _keys.map((e) => base64Encode(utf8.encode(e))).toList();
-    _secureStorage.saveKeys(encodedData);
+    await _secureStorage.saveKeys(encodedData);
   }
 
   Future<void> _getKeys() async {
     final decodedData = await _secureStorage.readKeys();
-    _keys = decodedData.map((e) => utf8.decode(base64Decode(e))).toList();
+    _keys = decodedData.map((e) => utf8.decode(base64Decode(e))).toSet();
```

Comment:
Do we want the keys to be cached instead of re-read all the time?

Reply:
I was thinking the same, but what if the storage_inspector changes that value? We do want to have te latest version everytime we want to use it right?

Reply:
Storage inspector has immutable keys for now ;) But yes, this probably should be fine